arch/Kconfig

```diff
@@ -357,11 +357,11 @@ config SYS_POWER_LOW_POWER_STATES_SUPPORTED
 	  This option signifies that the target supports the SYS_POWER_LOW_POWER_STATES
 	  configuration option.
```

Comment:
We've changed the meaning of `SYS_POWER_LOW_POWER_STATES_SUPPORTED`. Before it indicated that a given SoC supports low power and user is free to provide an implementation of `sys_suspend` function. Now that a power management subsystem implements `sys_suspend` the `SYS_POWER_LOW_POWER_STATES_SUPPORTED` indicates that the functionality is actually implemented by the SoC. There are multiple SoCs which do not implement low power support however they do
```
...
soc/arm/st_stm32/stm32l4/Kconfig.series:	select SYS_POWER_LOW_POWER_STATES_SUPPORTED
soc/arm/st_stm32/stm32f2/Kconfig.series:	select SYS_POWER_LOW_POWER_STATES_SUPPORTED
soc/arm/atmel_sam/same70/Kconfig.series:	select SYS_POWER_LOW_POWER_STATES_SUPPORTED
...
```
These should be removed. I.e. only SoCs that provide `sys_set_power_state()` should select `SYS_POWER_LOW_POWER_STATES_SUPPORTED` or `SYS_POWER_DEEP_SLEEP_STATES_SUPPORTED`.

Reply:
I will check which SoC provides sys_set_power_state() interface and update Kconfigs.
Thank you for catching that.